### ISSUE TITLE
Mapping PE reads that overlap

### ISSUE THREAD

**Issue by apc88** [FelixKrueger/Bismark#141]
Hi Felix,

I seem to be having trouble mapping 75bp PE reads that overlap. I trim for adapters using trimmomatic, which gives the following fragment length distribution:

![image](https://user-images.githubusercontent.com/24555120/32008736-f7313778-b97a-11e7-8a4e-2e41ab6c0eac.png) . As you can see, there is a drastic drop in mapped reads with length 60-75bp.

I tried using TrimGalore as well, but the length profile showed that all inserts <75bp were removed.

This is a very similar problem to the one explained in this biostart problem: https://www.biostars.org/p/227852/ , although the tricks that were suggested did not improve my situation.
Any advice?

Thank you!

**Comment by FelixKrueger** [comment]
Hi @apc88,

Whenever I was contacted in the past about oddities about the insert sizes it turned out that it had to do with the tool that was counting the insert sizes. As long as you are using a recent version of Bismark (which aligns using `--dovetail` by default) and Bowtie2 there is no reason why you would see  a drop at certain fragment length.

Trim Galore does not remove sequences shorter than 75 bp unless this is specified explcitely. The command
`trim_galore --paired file1.fastq.gz file2.fastq.gz`

Will remove sequence < 20bp by default.

I would recommend you run this one more time, and then import the BAM files in paired-end mode into SeqMonk, and then click of Read Length Histogram. I would bet a small amount of money that the curve will look just fine.

Let me know how you get on. Best, Felix

**Comment by apc88** [comment]
Hi Felix, thank you for your prompt response. Unfortunately, I get the same problem with SeqMonk.

![image](https://user-images.githubusercontent.com/24555120/32023543-8dab011c-b9a7-11e7-95f8-d98e5f0e7679.png)

Since we performed 75bp reads and are losing mapability at that length, I assume there must be some sort of issue when the reads overlap.

Any advice is appreciated!

**Comment by FelixKrueger** [comment]
Hmm, that's interesting. Would you be able to [email](felix.krueger@babraham.ac.uk) me say 200K completely untrimmed paired-end reads (gzipped) and mention the genome you used so I could take a look tomorrow? Cheers, Felix

**Comment by FelixKrueger** [comment]
Hi @apc88,

Thanks for the files. I ran a few test on the data, here are my thoughts:

- The data is directional, but I found it weird that R1 seems to contain quite a bit of read-through adapter contamination while R2 doesn't have any detectable adapter sequence. Normally R1 and R2 behave very similarly in this regard. How were the libraries prepared, is this a special kit?

- R1 shows some spikes in the base content at the start, most noticably at base 5 (with ~12% C). Where does this sequence bias come from, maybe it is a result of the kit you used?

- default trimming using Trim Galore (with the command above) resulted  
 in ~ 30% mapping efficiency. When I trimmed off the first 10bp from the start of both R1 and R2 it increased to 40% with default params. The fact that 5' end trimming increases the mapping efficiency suggests that something is causing an unwanted bias at the start of reads which you might want to get rid of before mapping. Again, maybe the kit you used yields some further clues into what to do. 

- I imported the resulting BAM file into SeqMonk (the 10bp trimmed PE version this case), and the fragment lengths look like this:
![read lengths](https://user-images.githubusercontent.com/5870509/32050903-de650e30-ba52-11e7-9a9f-43a7f191c48a.png)

The untrimmed version looked the same. I am afraid I don't see a dip around the read length as you did in the plots above. Only fragments < 40bp are less frequent, but this is probably expected (library prep size selection and mapping issues).

The commands I used were simply:

`trim_galore --paired --clip_r1 10 --clip_r2 10 file1.fastq.gz file2.fastq.gz`
and
`bismark --genome /bi/scratch/Genomes/Human/GRCh37/ -1 Short_R1_val_1.fq.gz -2 Short_R2_val_2.fq.gz`

So yea in short: The default mapping procedure mentioned above seems to yield fairly normal looking results. Do let me know if you've got any questions. 
Cheers, Felix

**Comment by apc88** [comment]
Hi Felix,

Apologies for the delay in responding. It seems my issues are with the adapters and not Bismark. 

Thank you for your help!

Regards,

**Comment by FelixKrueger** [comment]
Excellent! All the best,

**Comment by taojinsheng** [comment]
Hi Felix,

I have a similar problem when I did adapter and low quality trimming with trim_galore and them mapped the trimmed reads with bismark.If you like,I can  email you say 200K completely untrimmed paired-end reads (gzipped).My reference genome is hg19.And my bismark version is v0.17.0.Thank a lot!
<img width="469" alt="1" src="https://user-images.githubusercontent.com/23255364/49746816-f3e9f180-fcdc-11e8-95bd-500c64d067eb.png">

**Comment by FelixKrueger** [comment]
Hi @taojinsheng,

what exactly is the problem in your case? Without having looked at its in great detail it appears that the  software you used for plotting the insert size introduced a gap at pretty much 150bp, and tries to over-compensate a few bp later. This just smells like a plotting bug... Please feel free to email me some 200K sequences so I can take a look tonight.

Cheers, Felix

**Comment by taojinsheng** [comment]
Hi Felix,

Thank you for your early reply.I had emailed you (felix.krueger@babraham.ac.uk) some sample data.

**Comment by taojinsheng** [comment]
Hi Felix,

I found that if I change "--stringency 4" to "--stringency 1",the  weird insert size distribution disappeared.

**Comment by FelixKrueger** [comment]
Excellent, so can I regard this issue as 'solved'?

**Comment by taojinsheng** [comment]
yes,thanks.I just wonder why setting "--stringency 4" could generate this weird insert size distribution.What do you think is the cause?

**Comment by FelixKrueger** [comment]
To be honest I can't really think of a anything that would cause this, apart from some kind of bug in the plotting. I am pretty sure if you trim the data with `--stringency 1 to 4`, and would look at the read length histograms of the paired-end BAM file in SeqMonk you would not see this weird spikes.

**Comment by alexpcheng** [comment]
@taojinsheng @FelixKrueger I just came across this thread once again, and wanted to suggest trimming reads by overlap. I noticed that some aligners (like bowtie2) do not handle perfectly overlapping reads, and that removing base pairs that overlap in one read can greatly improve reported TLENs. Not sure if this is directly applicable to these datasets, but thought it might be useful if this issue comes up again.
